### ISSUE TITLE
Other image sizes not deleted when attachment is deleted from s3 bucket

### ISSUE THREAD

**Issue by rogden** [deliciousbrains/wp-amazon-s3-and-cloudfront#232]
When I delete an attachment, only the main file is deleted. None of the other image sizes are deleted. I believe the issue is in this function below. The value of $path in my instance is '.'. Therefor, the Key that is returned for each size for is ./image-100x100.jpg etc.. This doesn't match the key on s3 since the key shouldn't have a ./ in it.

```php
function prepare_intermediate_images_to_remove( $post_id, &$objects, $path  ) {
		$intermediate_images = get_intermediate_image_sizes();

		foreach (  $intermediate_images as $size ) {
			if ( $intermediate = image_get_intermediate_size( $post_id, $size ) ) {
				$objects[] = array(
					'Key' => path_join( $path, $intermediate['file'] )
				);
			}
		}
	}
```

**Comment by dbolton-resolute** [comment]
+1

**Comment by A5hleyRich** [comment]
Fixed in 0.9.7.
